python/cudf/cudf/core/column_accessor.py

```diff
@@ -27,6 +27,7 @@ def __init__(self, data={}, multiindex=False, level_names=None):
             For a non-hierarchical index, a tuple of size 1
             may be passe.
         """
+        data = data or {}
```

Comment:
Would suggest checking `None` explicitly throughout

```suggestion
        if data is None:
            data = {}
```

Reply:
sure I'd be happy to change this. I'll do that right now for all of this.

If you don't mind...why do you prefer this pattern instead of `or`? Is it just a personal style preference, or is there a performance different that I'm not aware of?

Reply:
I just changed these to the `if` pattern in https://github.com/rapidsai/cudf/pull/6410/commits/e553db21f4765e4fd9838b8bb9f6b35b2dcf5e06

Reply:
I think it's because for example if `[]` is another valid value for `data`, `data or {}` will coerce it to `{}`.